src/load/migrate-database.lisp

```diff
@@ -255,6 +255,29 @@
     (setf (catalog-distribution-rules catalog)
           (citus-distribute-schema catalog distribute))))
 
+(defun optimize-table-copy-ordering (catalog)
+  "Return a list of tables to copy over in optimized order"
+  (let ((table-list (copy-list (table-list catalog)))
+        (view-list  (copy-list (view-list catalog))))
+    ;; when materialized views are not supported, view-list is empty here
+    (cond
+      ((notevery #'null (mapcar #'table-row-count-estimate table-list))
+           (flet ((compare-row-count-estimates (a b)
+                    (declare (type (or null fixnum) a b))
+                    (> (or a 0) (or b 0))))
+             (let ((sorted-table-list (sort table-list
+                                            #'compare-row-count-estimates
+                                            :key #'table-row-count-estimate)))
+               (log-message :notice
+                            "Processing tables in this order: ~{~a: ~d rows~^, ~}"
+                            (loop :for table
+                               :in (append sorted-table-list view-list)
+                               :append (list (format-table-name table)
+                                             (table-row-count-estimate table))))
+               (append sorted-table-list view-list))))
+      (t
+       (append table-list view-list)))))
```

Comment:
`table-list` and `view-list` are both fresh; we can use `nconc` instead of `append` here.